### PR TITLE
[16.0][MIG] - purchase_deposit

### PR DESCRIPTION
**Pull request by Hardik-OSI**

### CODE REVIEW

**Comment by minhthie** [comment]
lgtm
image

image

File: purchase_deposit/models/purchase.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    _inherit = "purchase.order"
+
+    def copy_data(self, default=None):
+        if default is None:
+            default = {}
+        default["order_line"] = [
+            (0, 0, line.copy_data()[0])
+            for line in self.order_line.filtered(lambda l: not l.is_deposit)
+        ]
+        return super(PurchaseOrder, self).copy_data(default)
```
Comment (rousseldenis) [inline purchase_deposit/models/purchase.py:18]:
```suggestion
        return super().copy_data(default)
```

File: purchase_deposit/models/purchase.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    _inherit = "purchase.order.line"
+
+    is_deposit = fields.Boolean(
+        string="Is a deposit payment",
+        help="Deposit payments are made when creating bills from a purhcase"
+        " order. They are not copied when duplicating a purchase order.",
+    )
+
+    def _prepare_account_move_line(self, move=False):
+        res = super(PurchaseOrderLine, self)._prepare_account_move_line(move=move)
```
Comment (rousseldenis) [inline purchase_deposit/models/purchase.py:31]:
```suggestion
        res = super()._prepare_account_move_line(move=move)
```

File: purchase_deposit/models/res_config_settings.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+# Copyright 2019 Ecosoft Co., Ltd., Kitti U. <kittiu@ecosoft.co.th>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = "res.config.settings"
+
+    default_purchase_deposit_product_id = fields.Many2one(
```
Comment (rousseldenis) [inline purchase_deposit/models/res_config_settings.py:11]: Shouldn't this depends on company as can be a different product per company ? @kittiu

File: purchase_deposit/tests/test_purchase_deposit.py [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+# Copyright 2019 Ecosoft Co., Ltd., Kitti U. <kittiu@ecosoft.co.th>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields
+from odoo.exceptions import UserError
+from odoo.tests.common import Form, TransactionCase
+
+
+class TestPurchaseDeposit(TransactionCase):
+    def setUp(self):
```
Comment (rousseldenis) [inline purchase_deposit/tests/test_purchase_deposit.py:11]: Could you change to classmethod ?

File: purchase_deposit/wizard/purchase_make_invoice_advance.py [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+        help="The amount to be invoiced in advance, taxes excluded.",
+    )
+    deposit_account_id = fields.Many2one(
+        comodel_name="account.account",
+        string="Expense Account",
+        domain=[("deprecated", "=", False)],
+        help="Account used for deposits",
+    )
+    deposit_taxes_id = fields.Many2many(
+        comodel_name="account.tax",
```
Comment (rousseldenis) [inline purchase_deposit/wizard/purchase_make_invoice_advance.py:43]: Could you change this to computed store=True readonly=False field and change the onchange method to compute one as Odoo will not support onchanges in further version?

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review

**Comment by Saran440** [comment]
please cherry-pick https://github.com/OCA/purchase-workflow/pull/1329

**Comment by RodrigoBM** [comment]
@Hardik-OSI Are you going to add the suggested changes or is this PR dead? 
and the pr does not pass the tests

**Comment by AungKoKoLin1997** [comment]
Superseded by https://github.com/OCA/purchase-workflow/pull/1853.

**Pull request closed without merging**